docs: add TSDoc comments interface-internal module

Khwahish29 opened the pull request:
## Title
<!---
The title of the PR will be the commit message of the merge commit, so please make sure it is descriptive enough.
We utilize the Conventional Commits specification for our commit messages. See <https://www.conventionalcommits.org/en/v1.0.0/#specification> for more information.
The commit tag types can be of one of the following: feat, fix, deps, refactor, chore, docs. See <https://github.com/libp2p/js-libp2p/blob/master/.github/workflows/main.yml#L235-L242>
The title must also be fewer than 72 characters long or it will fail the Semantic PR check. See <https://github.com/libp2p/js-libp2p/blob/master/.github/workflows/semantic-pull-request.yml>
--->
docs: Add TSDoc comments to @libp2p/interface-internal (#2113 )


## Description
This PR adds TSDoc comments to `@libp2p/interface-internal`, ensuring that all exported interfaces, types, and constants have proper documentation. This improves the generated API documentation and aligns with the project's documentation standards.


<!--
Please write a summary of your changes and why you made them.
Please include any relevant issues in here, for example:
Related https://github.com/libp2p/js-libp2p/issues/ABCD.
Fixes https://github.com/libp2p/js-libp2p/issues/XYZ.
-->

## Notes & open questions
- Please review if the added examples align with best practices.
- Let me know if any additional clarifications are needed.

<!--
Any notes, remarks or open questions you have to make about the PR which don't need to go into the final commit message.
-->

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

maschad commented on packages/interface-internal/src/connection-manager.ts:
```diff
@@ -3,6 +3,17 @@ import type { PeerMap } from '@libp2p/peer-collections'
+ * @packageDocumentation
+ *
+ * The `ConnectionManager` module handles managing connections between peers in a libp2p network.
+ * It provides methods for opening, closing, and querying connections, ensuring efficient
```
I think we can exclude the part about `ensuring efficient peer-to-peer communication.` 

Also we should probably include that this also provides methods for accessing the dial queue.

maschad commented on packages/interface-internal/src/connection-manager.ts:
```diff
@@ -34,10 +45,16 @@ export interface OpenConnectionOptions extends AbortOptions, ProgressOptions<Ope
 }
 
+/**
+ * Manages peer-to-peer connections within a libp2p network.
```
Specifically this manages the dialling, opening and closing of connections.

maschad commented on packages/interface-internal/src/random-walk.ts:
````diff
@@ -1,13 +1,46 @@
+ *
+ * @example
+ * ```typescript
+ * import { randomWalk } from '@libp2p/random-walk'
````
This is an inaccurate example, given these are internal interfaces I don't think we need examples  regardless.

maschad commented on packages/interface-internal/src/transport-manager.ts:
```diff
@@ -2,13 +2,41 @@ import type { AbortOptions, Connection, Listener, Transport, TransportManagerDia
+ * @packageDocumentation
+ *
+ * The `TransportManager` module handles the management of transport protocols in a libp2p network.
+ * It is responsible for adding, removing, and managing transport connections used for peer communication.
```
It's responsible for managing the transports themselves - dialling, querying addresses and listening.

maschad reviewed (changes requested):
Great work here @Khwahish29 . I've left some comments.

My personal opinion is that we don't need examples for the applicability these internal interfaces, but interested to see @achingbrain 's opinion.

Khwahish29 commented:
> Great work here @Khwahish29 . I've left some comments.
> 
> My personal opinion is that we don't need examples for the applicability these internal interfaces, but interested to see @achingbrain 's opinion.

Hey @maschad! Took your feedback and updated the TSDoc comments for @libp2p/interface-internal
- Removed unnecessary examples.
- Cleaned up descriptions to keep things clear and consistent.

Really appreciate the thoughtful feedback! Let me know if there’s anything else to tweak. 🚀

achingbrain commented on packages/interface-internal/src/address-manager.ts:
```diff
@@ -1,5 +1,13 @@
 import type { Multiaddr } from '@multiformats/multiaddr'
 
+/**
```
I don't think we need `@packageDocumentation` tags for each file here.

From the [docs](https://tsdoc.org/pages/tags/packagedocumentation/):

> Used to indicate a doc comment that describes an entire NPM package (as opposed to an individual API item belonging to that package)

The text from the `@packageDocumentation` tags you've added could be used in the tag describing the main interface exported from each file instead.

Khwahish29 replied:
Thanks for the clarification! 🙌
I’ve moved the `@packageDocumentation` tags. Let me know if this looks good or if there’s anything else to adjust!

maschad commented on packages/interface-internal/src/connection-manager.ts:
```diff
@@ -34,49 +37,55 @@ export interface OpenConnectionOptions extends AbortOptions, ProgressOptions<Ope
   initiator?: boolean
 }
 
+/**
```
Based on my understanding of @achingbrain 's [comment](https://github.com/libp2p/js-libp2p/pull/2949#discussion_r1948420694) , we only need one `@packageDocumentation` for this `interface-internal` module, and we can just apply this explanatory comments to each interface exported, without the tag.

Khwahish29 replied:
Thanks for the clarification, @maschad ! 🙌

I’ve updated the PR to include only one `@packageDocumentation` tag for the module and moved the explanatory comments to the respective exported interfaces without the tag. Let me know if this aligns with what you had in mind or if any further adjustments are needed. Appreciate the feedback!

maschad reviewed (approved):
Thanks for this @Khwahish29 this LGTM

achingbrain commented on packages/interface-internal/src/address-manager.ts:
```diff
@@ -57,49 +57,70 @@ export interface ConfirmAddressOptions {
   type?: AddressType
 }
 
+/**
+ * The `AddressManager` module provides an interface for managing peer addresses
+ * in libp2p. It supports handling multiple types of addresses, verifying their validity,
+ * and storing mappings between internal and external addresses.
+ */
+/**
+ * Interface for managing peer addresses in libp2p.
+ */
+
 export interface AddressManager {
```
This needs to be one comment, not two, otherwise only the second will appear in the generated docs.

You can check this by running `npm run docs` in the root of the repo and looking at the HTML that appears in the `.docs` folder.

```suggestion
/**
 * The `AddressManager` provides an interface for managing peer addresses
 * in libp2p. It supports handling multiple types of addresses, verifying their validity,
 * and storing mappings between internal and external addresses.
 */
export interface AddressManager {
```

achingbrain commented on packages/interface-internal/src/connection-manager.ts:
```diff
@@ -34,49 +37,53 @@ export interface OpenConnectionOptions extends AbortOptions, ProgressOptions<Ope
   initiator?: boolean
 }
 
+/**
+ * The `ConnectionManager` module handles managing connections between peers in a libp2p network.
+ * It provides methods for opening, closing, and querying connections.This also provides methods
+ * for accessing the dial queue.
+ */
+/**
+ * Manages the dialing, opening, and closing of connections.
+ */
+
```
Must be a single comment, not two:

```suggestion
/**
 * The `ConnectionManager` handles managing connections between peers in a libp2p network.
 * It provides methods for opening, closing, and querying connections.This also provides methods
 * for accessing the dial queue.
 */
```

achingbrain commented on packages/interface-internal/src/random-walk.ts:
```diff
@@ -1,13 +1,25 @@
 /**
- * RandomWalk finds random peers on the network and dials them. Use this after
- * registering a Topology if you need to discover common network services.
+ * The `RandomWalk` module facilitates peer discovery by randomly finding and dialing peers 
+ * on the libp2p network. It is useful in conjunction with a registered `Topology` to 
+ * discover common network services.
+ */
+
+/**
+ * The `RandomWalk` interface provides a mechanism for discovering and connecting to
+ * random peers on the libp2p network.
```
Must be a single comment, not two.

Also, the docs were wrong here - RandomWalk only discovers peers, it doesn't dial them.

```suggestion
* The `RandomWalk` component uses the libp2p peer routing to find arbitrary
 * network peers. Consumers may then dial these peers, causing the Identify
 * protocol to run and any registered topologies to be notified of their
 * supported protocols.
```

achingbrain commented on packages/interface-internal/src/registrar.ts:
```diff
@@ -1,5 +1,10 @@
 import type { StreamHandler, StreamHandlerOptions, StreamHandlerRecord, Topology, IncomingStreamData } from '@libp2p/interface'
 
+/**
```
I'm not sure this comment is necessary - these are named exports so will get their own documentation blocks.

Please can you check the output of `npm run docs` (look in the `.docs` folder) and delete this comment if it doesn't appear anywhere.

achingbrain commented on packages/interface-internal/src/registrar.ts:
```diff
@@ -21,46 +26,75 @@ export type {
    */
   StreamHandlerRecord
 }
+/**
+ * The `Registrar` module provides an interface for managing protocol handlers
+ * and topologies in a libp2p network. It enables registering and managing
+ * protocol-specific handlers, ensuring efficient peer-to-peer communication.
+ */
+/**
+ * The `Registrar` interface allows modules to register, manage, and remove
+ * protocol handlers and topology discovery mechanisms in libp2p.
+ */
 
```
Must be a single comment, not two.

Also we can be a bit more specific in the comment:

```suggestion
/**
 * The `Registrar` provides an interface for registering protocol handlers -
 * these are invoked when remote peers open streams on the local node with the
 * corresponding protocol name.
 *
 * It also allows configuring network topologies for a given protocol(s). The
 * topology callbacks are invoked when a peer that supports those protocols
 * connects or disconnects.
 *
 * The Identify protocol must be configured on the current node for topologies
 * to function.
 */
```

achingbrain commented on packages/interface-internal/src/registrar.ts:
```diff
@@ -21,46 +26,75 @@ export type {
-   * passed protocol.
+   * Register a topology handler for a specific protocol.
    *
-   * An id will be returned that can later be used to unregister the
```
This part about the id being used to unregister the topology later is important, please do not delete it.

achingbrain commented on packages/interface-internal/src/transport-manager.ts:
```diff
@@ -2,13 +2,26 @@ import type { AbortOptions, Connection, Listener, Transport, TransportManagerDia
 export interface TransportManagerDialOptions extends AbortOptions, ProgressOptions<TransportManagerDialProgressEvents> {
 
 }
+/**
+ * The `TransportManager` module handles the management of transport protocols in a libp2p network.
+ * It is responsible for managing the transport themselves - dialling, querying addresses and listening.
+ */
+
+/**
+ * The `TransportManager` interface manages available transports for dialing and listening in a libp2p node.
+ */
 
```
Must be a single comment, not two.

Also `protocols` is a loaded word here, since it is used to refer to protocol streams.  Better to just call transports "transports" or "network transports" and not "transport protocols".

Also, also, it's probably worth noting that consumers should generally prefer to open connections using the connection manager.

```suggestion

/**
 * The `TransportManager` handles the management of network transports, allowing
 * opening connections or listening using specific transports, etc.
 *
 * This is a low-level component - any connections opened will not be managed by
 * the `ConnectionManager` or included in any configured connection limits, etc.
 *
 * Most consumers will call `openConnection` on the `ConnectionManager` instead.
 */
```

Khwahish29 replied:
@achingbrain I've addressed the latest round of feedback—please let me know if there's anything else that needs further improvement.
I truly appreciate the time and effort you’ve put into reviewing my work and ensuring it meets the right standards :)

achingbrain reviewed (changes requested):
Good stuff, comments inline.

achingbrain commented on packages/interface-internal/src/transport-manager.ts:
```diff
@@ -2,13 +2,28 @@ import type { AbortOptions, Connection, Listener, Transport, TransportManagerDia
 export interface TransportManagerDialOptions extends AbortOptions, ProgressOptions<TransportManagerDialProgressEvents> {
 
 }
+/**
+ * The `TransportManager` handles the management of network transports, allowing
+ * opening connections or listening using specific transports, etc.
+ *
+ * This is a low-level component - any connections opened will not be managed by
+ * the `ConnectionManager` or included in any configured connection limits, etc.
+ *
+ * Most consumers will call `openConnection` on the `ConnectionManager` instead.
+ */
+
 
```
```suggestion

/**
 * The `TransportManager` handles the management of network transports, allowing
 * opening connections or listening using specific transports, etc.
 *
 * This is a low-level component - any connections opened will not be managed by
 * the `ConnectionManager` or included in any configured connection limits, etc.
 *
 * Most consumers will call `openConnection` on the `ConnectionManager` instead.
 */
```

achingbrain commented on packages/interface-internal/src/address-manager.ts:
```diff
@@ -57,49 +57,66 @@ export interface ConfirmAddressOptions {
   type?: AddressType
 }
 
+/**
+ * The `AddressManager` module provides an interface for managing peer addresses
+ * in libp2p. It supports handling multiple types of addresses, verifying their validity,
+ * and storing mappings between internal and external addresses.
+ */
```
```suggestion
/**
 * The `AddressManager` provides an interface for managing peer addresses
 * in libp2p. It supports handling multiple types of addresses, verifying their validity,
 * and storing mappings between internal and external addresses.
 */
```
